pkg/kubectl/cmd/run.go

```diff
@@ -466,16 +471,41 @@ func (o *RunOptions) removeCreatedObjects(f cmdutil.Factory, createdObjects []*R
 }
 
 // waitForPod watches the given pod until the exitCondition is true
-func waitForPod(podClient coreclient.PodsGetter, ns, name string, exitCondition watch.ConditionFunc) (*api.Pod, error) {
-	w, err := podClient.Pods(ns).Watch(metav1.SingleObject(metav1.ObjectMeta{Name: name}))
-	if err != nil {
-		return nil, err
+func waitForPod(podClient coreclient.PodsGetter, ns, name string, exitCondition watchtools.ConditionFunc) (*api.Pod, error) {
+	// TODO: expose the timeout
```

Comment:
Don't leave a todo, expose it.

Reply:
out of scope for this change, if it wasn't till then - it's not the intend of these changes to fix the interface, just the machinery